### PR TITLE
bugfix of parsing error

### PR DESCRIPTION
**Pull request by chanmaoganda**
This is my implemetion for parsing diff types of NcmId

### CODE REVIEW

**Comment by chanmaoganda** [comment]
this version can works well with most test cases
But i only tested ncm files, other forms were not tested
Perhaps it would behave correctly

File: crates/ncmdump/src/ncmdump.rs [hunk 1]
```diff
@@ -55,6 +84,39 @@ where
...
+            NcmId::String(s) => {
+                if s.is_empty() {
+                    return None;
+                }
+                s.parse().ok()
+            },
+            NcmId::Integer(num) => Some(num),
+        }
+    }
+}
```
Comment (iqiziqi) [inline crates/ncmdump/src/ncmdump.rs]: perhaps we should resolve error by better way. For example use `Resule` and `Errors`.

**Comment by iqiziqi** [comment]
Sorry I didn't bring my computer, I'll deal with it tomorrow

File: crates/ncmdump/src/ncmdump.rs (hunk 1 above)
Comment (chanmaoganda) [reply crates/ncmdump/src/ncmdump.rs:118]:
```suggestion
impl NcmId {
    pub fn get_id(self) -> Option<u64> {
        match self {
            NcmId::String(s) => {
                if s.is_empty() {
                    return None;
                }
                s.parse().ok()
            },
            NcmId::Integer(num) => Some(num),
        }
    }
}
```

Comment (chanmaoganda) [reply crates/ncmdump/src/ncmdump.rs]: I don't fully understand your mind, can u explain it to me?

Comment (iqiziqi) [reply crates/ncmdump/src/ncmdump.rs]: The `ncmId::String("")` will make a panic.

Comment (chanmaoganda) [reply crates/ncmdump/src/ncmdump.rs]: u mean this function should panic when parsing NcmId::String("")?
i am not a native english speaker, apologies for my poor english😖

**Comment by chanmaoganda** [comment]
but in one test case like below will panic!
perhaps mvId with empty string should be normal??

```json
{
   "musicId": "2056228066",
   "musicName": "Eternal regret",
   "artist": [
      [
         "AnRain安林",
         "35516167"
      ]
   ],
   "albumId": "167614757",
   "album": "Eternal regret",
   "albumPicDocId": "109951168681303606",
   "albumPic": "http://p4.music.126.net/XfUBgbH9RDQqBcStBVa-Pw==/109951168681303606.jpg",
   "bitrate": 320000,
   "mp3DocId": "d27e546865e86910e1da2ccbefd00a15",
   "duration": 186741,
   "mvId": "",
   "alias": [],
   "transNames": [],
   "format": "mp3",
   "fee": 8,
   "volumeDelta": -8.0989,
   "privilege": {
      "flag": 1544198
   }
}
```

**Comment by chanmaoganda** [comment]
That will be great!😊

**Pull request merged**